Post Embedded World 2019 Cleanups

borancar opened the pull request:
- Refactor the XEmacLite driver to avoid unnecessary warnings, also make it more generic so it only needs an MMIO address to function and so we can have multiple devices.
- Fix the ASN warning in zone4 (RoT zone)
- Disable wolfSSL memory callbacks as we don't need them anymore (we're not measuring memory usage anymore)
- Fix a compilation warning by skipping wolfSSL misc.c (only needed when NOINLINE)
- Update multizone dependency (new fixes in the kernel, configuratior and the libraries)
- Use the new ECALL_SEND/ECALL_RECV semantics (we get a return whether successful or not)
- Use the X300 JEDEC ID in zone CLIs so we can report Hex Five as the vendor
- Add an ignore for .envrc, developers can use it to define a local override for IPADDR
- Increase msg queue buffer to 128 bytes now that we're transferring 16 bytes at a time
- Rename mzmsg to just msg, and clarify when channel is used for CLI access

cgarlati commented:
@borancar Issues:

1) stats go above 10ms when telnet is connected (was way below before) 
2) utilization stays above tick time 10ms even when the telnet is closed
3) led stops blinking and becomes green solid pressing btn0 twice *after* telnet disconnect. Resume blinking when telnet is reconnected.

borancar commented:
> @borancar Issues:
> 
>     1. stats go above 10ms when telnet is connected (was way below before)

Fixed in https://github.com/hex-five/multizone-secure-iot-stack/pull/56/commits/53b64bd8006d3e3c21fca8652256d1cca13a4b12

>     2. utilization stays above tick time 10ms even when the telnet is closed

Fixed in https://github.com/hex-five/multizone-secure-iot-stack/pull/56/commits/53b64bd8006d3e3c21fca8652256d1cca13a4b12

>     3. led stops blinking and becomes green solid pressing btn0 twice _after_ telnet disconnect. Resume blinking when telnet is reconnected.

That's the case on master, too, since mzmsg_write is blocking (will keep yielding until the whole message is transmitted). I would propose merging this PR in and doing those fixes in a separate refactor of FreeRTOS.

cgarlati commented:
@borancar please file an issue to track 3. led stops blinking and becomes green solid pressing btn0 twice _after_ telnet disconnect. Resume blinking when telnet is reconnected.
